File: mobile_app/lib/features/admin/settings/domain/entities/setting.dart
class Setting {
  final String key;
  final String value;
  final String category;
  final String label;
  final String? description;
  final String fieldType;
  final List<String>? options;
  final String defaultValue;
  final int order;

  const Setting({
    required this.key,
    required this.value,
    required this.category,
    required this.label,
    this.description,
    required this.fieldType,
    this.options,
    required this.defaultValue,
    this.order = 0,
  });

  bool get isChanged => value != defaultValue;
}

class SettingsGroup {
  final String category;
  final String label;
  final List<Setting> settings;

  const SettingsGroup({
    required this.category,
    required this.label,
    required this.settings,
  });
}

class AppSettings {
  final List<SettingsGroup> groups;
  final Map<String, String> raw;

  const AppSettings({required this.groups, required this.raw});

  String? getValue(String key) => raw[key];
}

class VersionInfo {
  final String applicationVersion;
  final String backendVersion;
  final String environment;

  const VersionInfo({
    required this.applicationVersion,
    required this.backendVersion,
    required this.environment,
  });
}

class DatabaseStatus {
  final String status;
  final int totalTables;
  final int totalRecords;

  const DatabaseStatus({
    required this.status,
    required this.totalTables,
    required this.totalRecords,
  });

  bool get isConnected => status == 'connected';
}

File: 03_Backend/app/schemas/setting.py
from typing import Any

from pydantic import Field

from app.schemas.base import BaseSchema


class SettingUpdate(BaseSchema):
    value: str = Field(..., min_length=0, description="Setting value")


class SettingsBatchUpdate(BaseSchema):
    settings: dict[str, str] = Field(
        ..., description="Key-value map of settings to update"
    )


class SettingResponse(BaseSchema):
    key: str
    value: str
    category: str
    label: str
    description: str | None = None
    field_type: str
    options: list[str] | None = None
    default_value: str
    order: int = 0


class SettingsGroupResponse(BaseSchema):
    category: str
    label: str
    settings: list[SettingResponse]


class SettingsResponse(BaseSchema):
    groups: list[SettingsGroupResponse]
    raw: dict[str, str]


class DatabaseStatusResponse(BaseSchema):
    status: str
    total_tables: int
    total_records: int
    size_mb: float | None = None


class VersionInfoResponse(BaseSchema):
    application_version: str
    backend_version: str
    environment: str
